### PR TITLE
ns2.1 and TryResolveStateMachineMethod

### PR DESCRIPTION
**Pull request by bruno-garcia**
it's an easy change to discuss while looking at the change.

* Makes sense to have ns2.1?
* The goal is to add `AsyncIteratorStateMachineAttribute`

Nice CI is there! 🚀

### CODE REVIEW

**Comment by benaadams** [comment]
```
Feature 'nullable reference types' is not available in C# 7.3. Please use language version 8.0 or greater.
```
```
Feature 'static local functions' is not available in C# 7.3. Please use language version 8.0 or greater.
```

Is lang `latest` conditional based on TFM?

**Comment by bruno-garcia** [comment]
It "works on my machine" as it builds here.
~I assume it requires net5 installed.~

I see CI has it:
```
      - name: Install .NET SDK 5.0
        uses: actions/setup-dotnet@v1
        with:
          dotnet-version: '5.0.x'
```

**Comment by bruno-garcia** [comment]
I'll remove it

**Comment by bruno-garcia** [comment]
Oddly enough only failed on Ubuntu

File: Directory.Build.props [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ﻿<Project>
   <PropertyGroup>
+    <LangVersion>8.0</LangVersion>
```
Comment (benaadams) [inline Directory.Build.props]: Problematic for the fsharp project? (as that's only `4.7`...)

I think `latest` and `preview` work for that too

Comment (bruno-garcia) [reply Directory.Build.props]: Good point. Pushed, let's see.

**Comment by benaadams** [comment]
I'm going to check my casing in another project I have that was having the same issue with nullability 😅

**Comment by benaadams** [comment]
Also a test to show its doing something? (Resolving the Iterator)

File: Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿<Project>
+  <PropertyGroup>
+    <LangVersion>latest</LangVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetFramework)' != 'netstandard2.0' and !$(TargetFramework.StartsWith('netcoreapp2')) and !$(TargetFramework.StartsWith('net4'))">
```
Comment (bruno-garcia) [inline Directory.Build.props:5]: adding `net6` or any other tfm in the future would include `HAS_ASYNC_ENUMERATOR` without anyone having to remember opting-in.

File: README.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Ben.Demystifier
 
 [![NuGet version (Ben.Demystifier)](https://img.shields.io/nuget/v/Ben.Demystifier.svg?style=flat-square)](https://www.nuget.org/packages/Ben.Demystifier/)
-[![Build status](https://ci.appveyor.com/api/projects/status/rlhygchfph287fq0?svg=true)](https://ci.appveyor.com/project/benaadams/ben-demystifier)
+[![build](https://github.com/benaadams/Ben.Demystifier/workflows/Demystifier%20PR%20Build/badge.svg)](https://github.com/benaadams/Ben.Demystifier/actions)
```
Comment (bruno-garcia) [inline README.md]: oh why is this here. oh git

**Review by benaadams — APPROVED** [review]

**Comment by benaadams** [comment]
Thank you :)
